File: check_syntax.js
const fs = require('fs');

try {
    const html = fs.readFileSync('index.html', 'utf8');
    const scripts = html.match(/<script>([\s\S]*?)<\/script>/g);
    
    if (scripts && scripts.length > 0) {
        const lastScript = scripts[scripts.length - 1].replace(/<\/?script>/g, '');
        fs.writeFileSync('temp.js', lastScript);
        console.log('Script extracted to temp.js');
    } else {
        console.log('No scripts found');
    }
} catch (e) {
    console.error('Error:', e);
}

File: check_syntax.py
import re
import ast

def check_js_syntax():
    try:
        with open('index.html', 'r', encoding='utf-8') as f:
            html = f.read()
            
        scripts = re.findall(r'<script>(.*?)</script>', html, re.DOTALL)
        if not scripts:
            print("No scripts found")
            return
            
        js_code = scripts[-1]
        
        # Save to temp file and try to parse it
        with open('temp.js', 'w', encoding='utf-8') as f:
            f.write(js_code)
            
        print("JS code extracted to temp.js")
        
    except Exception as e:
        print(f"Error: {e}")

if __name__ == "__main__":
    check_js_syntax()